File: DONGUK/프로그래머스_WEEK 4/소수 찾기.py
from itertools import permutations

def isPrime(n):
    if n == 0 or n == 1:
        return False
    
    for i in range(2, n):
        if n % i == 0:
            return False
    
    return True


def solution(numbers):
    answer = []
    
    nums = list(map(int, numbers))
    
    for num in nums:
        if isPrime(num) and num not in answer:
            answer.append(num)
            
    
    nums = list(map(str, numbers))
    for i in range(2, len(nums) + 1):
        data_list = list(permutations(nums, i))
        for data in data_list:
            num = int(''.join(data))
            
            if num not in answer and isPrime(num):
                answer.append(num)
                
    answer = len(answer)
    return answer

File: JINSEOP/프로그래머스_WEEK15/소수 찾기.cpp
#include <string>
#include <vector>
using namespace std;

bool seive[1000001] = {0,};
int seive_of_eratosthenes(int n){
    int cnt = 0;
    
    for(int i = 2; i <= n; i++){
        if(seive[i]) continue;
        
        for(int j = i*2; j <= n; j = j+i)
                seive[j] = true;
    }
    
    for(int i = 2; i <= n; i++)
        if(!seive[i])
            cnt++;
    
    return cnt;
}

int solution(int n) {
    int answer = 0;

    answer = seive_of_eratosthenes(n);
    return answer;
}

// 에라토스테네스의 체.
